### PR TITLE
fix: 114: handle ParseException from PBJ Codec.parse()

### PR DESCRIPTION
**Pull request by anthony-swirldslabs**
**Description**:
* Upgrading PBJ dependency to 0.7.14
* Replacing handling of the `IOException` thrown by methods in the Codec interface with equivalent handling of the new `ParseException` introduced in PBJ 0.7.14

**Related issue(s)**:

Fixes #https://github.com/hashgraph/pbj/issues/114

**Notes for reviewer**:
The Gradle `test` target passes in each of the modified repositories.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/codecs/EntityNumCodec.java
Comment (jsync-swirlds) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/codecs/EntityNumCodec.java:1]: hmmm, I'm not sure this class even has a purpose anymore.
Probably it should be removed, but perhaps not in this PR.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java [hunk 1]
```diff
@@ -70,7 +71,7 @@ public static <T> int writeToStream(
      */
     @NonNull
     public static <T> T readFromStream(@NonNull final InputStream in, @NonNull final Codec<T> codec)
```
[the remark this answers is not in the document]
Comment (anthony-swirldslabs) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java:73]: That's a good catch. I missed the semantics of this particular method/class while I was preparing this massive change. Let me fix this in the next commit.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/codecs/EntityNumCodec.java
Comment (anthony-swirldslabs) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/codecs/EntityNumCodec.java:1]: I'm not sure about this. Please file an issue against the team that owns this particular class to decide this.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (anthony-swirldslabs) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/StateUtils.java]: I just fixed that in the latest commit. Please have a look.

**Review by jsync-swirlds — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]
LGTM wrt RE owned files

**Review by povolev15 — APPROVED** [review]
LGTM

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java [hunk 2]
```diff
@@ -297,15 +298,15 @@ public void handleQuery(@NonNull final Bytes requestBuffer, @NonNull final Buffe
...
-            // This should technically not be possible. The data buffer supplied
-            // is either based on a byte[] or a byte buffer, in both cases all data
-            // is available and a generic IO exception shouldn't happen. If it does,
-            // it indicates the data could not be parsed, but for a reason other than
-            // those causing an MalformedProtobufException or UnknownFieldException.
-            logger.warn("Unexpected IO exception while parsing protobuf", e);
+        } catch (ParseException e) {
+            switch (e.getCause()) {
+                case MalformedProtobufException ex:
+                    break;
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:304]: We should probably throw exception in both these cases here to keep same behavior
```
throw new StatusRuntimeException(Status.INVALID_ARGUMENT);
```

Comment (anthony-swirldslabs) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:304]: This exception is thrown in all of these cases right below the `switch()` at line 310. This `switch` statement is only used to decide if we want to log a warning. It's coded to implement the exact same logic as was present before with individual `catch()` clauses.

**Review by Neeharika-Sompalli — APPROVED** [review]
`hedera-base` changes LGTM , except for one comment. Tyhanks @anthony-swirldslabs

**Review by Nana-EC — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:304]: ah yes ! makes sense. Thanks @anthony-swirldslabs

**Pull request merged**